File: version/1.18.2/src/main/java/com/github/sorusclient/client/adapter/v1_18_2/TextClickEventImpl.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter.v1_18_2;

import com.github.sorusclient.client.adapter.ClickEventAction;
import com.github.sorusclient.client.adapter.ITextClickEvent;
import v1_18_2.net.minecraft.text.ClickEvent;

public class TextClickEventImpl implements ITextClickEvent {

    private final String value;
    private final ClickEventAction action;

    public TextClickEventImpl(ClickEvent clickEvent) {
        value = clickEvent.getValue();
        action = switch (clickEvent.getAction()) {
            case SUGGEST_COMMAND -> ClickEventAction.SUGGEST_COMMAND;
            case RUN_COMMAND -> ClickEventAction.RUN_COMMAND;
            case OPEN_URL -> ClickEventAction.OPEN_URL;
            default -> null;
        };
    }

    @Override
    public String getValue() {
        return value;
    }

    @Override
    public ClickEventAction getAction() {
        return action;
    }

}

File: version/1.8.9/src/main/kotlin/com/github/sorusclient/client/adapter/v1_8_9/TextClickEventImpl.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter.v1_8_9

import com.github.sorusclient.client.adapter.ClickEventAction
import com.github.sorusclient.client.adapter.ITextClickEvent
import v1_8_9.net.minecraft.text.ClickEvent

class TextClickEventImpl(clickEvent: ClickEvent) : ITextClickEvent {
    override val value: String
    override val action: ClickEventAction

    init {
        value = clickEvent.value
        this.action = when (clickEvent.action) {
            ClickEvent.Action.SUGGEST_COMMAND -> ClickEventAction.SUGGEST_COMMAND
            ClickEvent.Action.RUN_COMMAND -> ClickEventAction.RUN_COMMAND
            ClickEvent.Action.OPEN_URL -> ClickEventAction.OPEN_URL
            else -> null!!
        }
    }

}